rotkehlchen/externalapis/monerium.py

```diff
@@ -37,47 +49,53 @@ class Monerium:
     """
 
-    def __init__(self, database: 'DBHandler', user: str, password: str) -> None:
+    def __init__(self, database: 'DBHandler') -> None:
         self.database = database
         self.session = create_session()
-        self.user = user
-        self.password = password
         set_user_agent(self.session)
+        self.oauth_client = MoneriumOAuthClient(database=self.database, session=self.session)
 
     def _query(
             self,
-            endpoint: Literal['orders'],
+            endpoint: str,
             params: dict[str, Any] | None = None,
-    ) -> list[dict[str, Any]]:
-        """Query a monerium API endpoint with basic authentication"""
-        querystr = 'https://api.monerium.app/' + endpoint
-        log.debug(f'Querying monerium API  {querystr}')
+    ) -> dict[str, Any]:
+        """Query a monerium API endpoint using OAuth authenticated requests.
+        May raise:
+            - AuthenticationError
+            - RemoteError
+        """
+        if not self.oauth_client.is_authenticated():
+            raise AuthenticationError('Monerium not authenticated')
+
+        log.debug(f'Querying monerium API {endpoint}')
         timeout = CachedSettings().get_timeout_tuple()
-        try:
-            response = self.session.get(
-                url=querystr,
-                params=params,
-                timeout=timeout,
-                auth=(self.user, self.password),
-            )
-        except requests.exceptions.RequestException as e:
-            raise RemoteError(f'Querying {querystr} failed due to {e!s}') from e
+        return self.oauth_client.get(endpoint, params, timeout)
 
-        if response.status_code != 200:
-            raise RemoteError(
-                f'Monerium API request {response.url} failed '
-                f'with HTTP status code {response.status_code} and text '
-                f'{response.text}',
-            )
+    def is_authenticated(self) -> bool:
+        return self.oauth_client.is_authenticated()
 
-        try:
-            json_ret = jsonloads_list(response.text)
-        except JSONDecodeError as e:
-            raise RemoteError(
-                f'Monerium API returned invalid JSON response: {response.text}',
-            ) from e
+    def get_oauth_status(self) -> dict[str, Any]:
+        return self.oauth_client.get_status()
 
-        return json_ret
+    def complete_oauth(
+            self,
+            access_token: str,
+            refresh_token: str,
+            expires_in: int,
+            client_id: str,
+            token_type: str = 'Bearer',  # noqa: S107
+    ) -> dict[str, Any]:
+        return self.oauth_client.complete_oauth(
+            access_token=access_token,
+            refresh_token=refresh_token,
+            expires_in=expires_in,
+            client_id=client_id,
+            token_type=token_type,
+        )
+
+    def disconnect(self) -> dict[str, Any]:
+        return self.oauth_client.disconnect()
```

Comment:
what does this do? invalidate the access & refresh token?

Reply:
deletes the token information from the db and clears the monerium object attributes